acvm_backend_barretenberg/src/scalar_mul.rs

```diff
@@ -1,21 +1,30 @@
 use common::acvm::FieldElement;
-use wasmer::Value;
 
 use super::{Barretenberg, FIELD_BYTES};
 
 impl Barretenberg {
-    pub fn fixed_base(&mut self, input: &FieldElement) -> (FieldElement, FieldElement) {
-        let lhs_ptr: usize = 0;
-        let result_ptr: usize = lhs_ptr + FIELD_BYTES;
+    pub(crate) fn fixed_base(&self, input: &FieldElement) -> (FieldElement, FieldElement) {
+        cfg_if::cfg_if! {
```

Comment:
I'm getting super lost in all the cfg_if branches. Can we just make the various implementations their own functions and just toggle them with the config macro?

Reply:
My concern about this was ensuring that we match interfaces. I've used private traits to address this in [1d2369e](https://github.com/noir-lang/aztec_backend/pull/117/commits/1d2369e0e91db1581452f3e242347e294d0c23d0)